report/src/main/webapp/app/components/compareScreens.directive.js

```diff
@@ -0,0 +1,648 @@
+/*
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(['angularAMD'], function (angularAMD) {
+  'use strict';
+  angularAMD.directive('aetCompareScreens', compareScreensDirective);
+
+  function compareScreensDirective() {
+    return {
+      restrict: 'A',
+      link: linkFunc
+    };
+  }
+
+  function linkFunc(scope) {
+    scope.advancedScreenComparison = function () {
+      var tab = document.querySelector('.test-tabs > .tab-content > .tab-pane.ng-scope.active');
+      var wrapper = tab.querySelector('.page-main .layout-compare');
+      var initedClass = 'screen-comparison-active';
+      var arrangeTimeout = 0;
+
+      function prepareMarkup(wrapper) {
+        wrapper.innerHTML = '</div><div class="difs"></div><div class="customMasks"><div class="customMask maskAtop"></div><div class="customMask maskAbot"></div><div class="customMask maskBtop"></div><div class="customMask maskBbot"></div><div class="customLabel labelA"><div class="ruler"></div><div class="text"></div></div><div class="customLabel labelB"><div class="ruler"></div><div class="text"></div></div></div><div class="canvas-wrapper"></div>';
+      }
+
+      if (!wrapper) {
+        console.warn('Can\'t find wrapper');
+        return;
+      } else if (wrapper.classList.contains(initedClass)) {
+        console.warn('Screen comparison already has already been initialized');
+        return;
+      } else {
+        wrapper.classList.add(initedClass);
+      }
+
+      var button = document.querySelector('.try-new');
```

Comment:
Could you please encapsulate logic in the lines below into a function, e.g. `markComparisonStarted` or smth. similar?

Reply:
Done